### PR TITLE
Singleplayer prototype implemented

### PR DESCRIPTION
**Pull request by MacMat01**
- Implemented CardTracker logic to save the time, card, player, enemy, round ad phase of each palyed card e4cddeba76607f2f69a68b4b6a38bfd2c7db29f0
- Implemented MatchManager [7a1f4df](https://github.com/SenseiBonsai2K/strategic-fruits/pull/23/commits/7a1f4df3ffc1e6c0715a7997fe7fe24da78fb1bd) to estabilish the matchup and the winner(or tie [078cee0](https://github.com/SenseiBonsai2K/strategic-fruits/pull/23/commits/078cee08e5e223f1aedeb57b2959ced3b7609919)) in each round
- Implemented the creation of the grid [d7cf943](https://github.com/SenseiBonsai2K/strategic-fruits/pull/23/commits/d7cf9436670c0f9af1be6c91077f5d63fad3e978)
- Implemented GridManager to fill the grid when a player win [db163ed](https://github.com/SenseiBonsai2K/strategic-fruits/pull/23/commits/db163edc31f60201e137d4d531775432f0553446)
- Implemented the calculation of the point and the final winner at the end of the game [91a912c](https://github.com/SenseiBonsai2K/strategic-fruits/pull/23/commits/91a912cf07f396331126844aba4e89f8d0823304)
- Implemented Backgound music and FX Sounds when a card is played and tokens are placed [e3c2095](https://github.com/SenseiBonsai2K/strategic-fruits/pull/23/commits/e3c2095b7366629c4600b1e065244e7fd0b98c6e)
- New sounds in Assets
- New textures and materials for cards, cells and token in Assets

### CODE REVIEW

File: Assets/Scripts/CardTracker.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        var timeElapsed = Time.time - _roundStartTime;
+        PlayedCards.Add((card, timeElapsed));
+
+        Debug.Log($"Card saved: {card.Rank} of {card.Suit}, Time elapsed: {timeElapsed}");
+
+        //TODO InfluxDb integration
+        // InfluxDBManager.WritePointOnDatabase(GameManager.CurrentPhase, GameManager.CurrentRound,
+        //     MatchManager.GetOpponent(GameManager.CurrentRound, card.Suit),
+        //     card.Suit, card.Rank, timeElapsed);
```
Comment (MacMat01) [inline Assets/Scripts/CardTracker.cs:35]: I would delete this comments since we won't be using InfluxDB

File: Assets/Scripts/FillHand.cs [hunk 2]
```diff
@@ -40,15 +40,20 @@ internal void Fill()
         // Generate the array of cards
         Cards = GenerateCard();
 
+        // Instantiate a card prefab for each card in the array
         // Instantiate a card prefab for each card in the array
```
Comment (MacMat01) [inline Assets/Scripts/FillHand.cs:44]: There seems to be two duplicated lines

File: Assets/Scripts/InfluxDBManager.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+//             .Field("Matchup", matchup)
+//             .Field("Rank", rank)
+//             .Field("Thinking Time", thinkingTime)
+//             .Timestamp(DateTime.UtcNow, WritePrecision.Ns);
+//
+//         using var writeApi = _client.GetWriteApi();
+//         writeApi.WritePoint(point);
+//         Debug.Log($"Data saved: Phase {phase}, Round {round}, Matchup {matchup}, Suit {suit}, Rank {rank}, Thinking Time {thinkingTime}");
+//     }
+// }
```
Comment (MacMat01) [inline Assets/Scripts/InfluxDBManager.cs:39]: I would delete this class for the reason explained earlier

File: Assets/Scripts/FillHand.cs (hunk 2 above)
Comment (SenseiBonsai2K) [reply Assets/Scripts/FillHand.cs]: Fixed 9329144a20154400c78b5891526ff5ffe8f16c9c

File: Assets/Scripts/InfluxDBManager.cs (hunk 3 above)
Comment (SenseiBonsai2K) [reply Assets/Scripts/InfluxDBManager.cs]: Fixed [b84c3fe](https://github.com/SenseiBonsai2K/strategic-fruits/pull/23/commits/b84c3fe8039bea4ddade53288d33b087ca10bbda)

File: Assets/Scripts/CardTracker.cs (hunk 1 above)
Comment (SenseiBonsai2K) [reply Assets/Scripts/CardTracker.cs]: Fixed [9019a6a](https://github.com/SenseiBonsai2K/strategic-fruits/pull/23/commits/9019a6a8fb44946d6fffda991f0640d4d76add08)

**Pull request merged**